wgpu/src/widget/canvas/path/builder.rs

```diff
@@ -133,11 +133,14 @@ impl Builder {
     ///
     /// [`Path`]: struct.Path.html
     #[inline]
-    pub fn rectangle(&mut self, p: Point, size: Size) {
-        self.move_to(p);
-        self.line_to(Point::new(p.x + size.width, p.y));
-        self.line_to(Point::new(p.x + size.width, p.y + size.height));
-        self.line_to(Point::new(p.x, p.y + size.height));
+    pub fn rectangle(&mut self, top_left: Point, size: Size) {
```

Comment:
this is totally pedantic, but generally code like this refers to the top left point as `origin`

Reply:
I have seen graphics APIs where rectangles are defined using the center or the bottom-left corner as the starting point. This is just an explicit way of documenting the API. The same way the `circle` method takes a `center` point.

This way, you can tell how to use this method by just looking at the method signature.